core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/JvmStartupOptimizerArchiveResultBuildItem.java

```diff
@@ -18,11 +18,11 @@ public final class AppCDSResultBuildItem extends SimpleBuildItem {
      */
     private final JvmStartupOptimizerArchiveType type;
 
-    public AppCDSResultBuildItem(Path appCDS) {
+    public JvmStartupOptimizerArchiveResultBuildItem(Path appCDS) {
         this(appCDS, JvmStartupOptimizerArchiveType.AppCDS);
     }
 
-    public AppCDSResultBuildItem(Path appCDS, JvmStartupOptimizerArchiveType type) {
+    public JvmStartupOptimizerArchiveResultBuildItem(Path appCDS, JvmStartupOptimizerArchiveType type) {
         this.appCDS = appCDS;
```

Comment:
The field name and the variable name need an adjustment.

Also the getter below.